### PR TITLE
[CI] Buildkite integration tests: Run integration tests on ARM

### PR DESCRIPTION
**Pull request by pazone**
## What does this PR do?
Enhances the buildkite tests pipeline by ARM tests. It uses the prebuilt Ubuntu 24.04 ARM image
Runs integration test groups separately. Allows independent reruns.
Tested here: [Example build](https://buildkite.com/elastic/elastic-agent-extended-testing-bk/builds/33#_). 

Note: It doesn't replace current integration tests. The new buildkite pipeline is not triggered by GitHub events

## Why is it important?
We have to test elastic agent on ARM architecture

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

- Closes https://github.com/elastic/ingest-dev/issues/4217
- Relates #123
- Requires #123
- Superseds #123

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by pchila — APPROVED** [review]
LGTM

**Comment by rowlandgeoff** [comment]
Where do we run the non-sudo tests?

**Comment by pazone** [comment]
> Where do we run the non-sudo tests?

I can add return non-sudo tests back to the pipeline. I'd do that in a separate PR

File: .buildkite/bk.integration.pipeline.yml [hunk 1]
```diff
@@ -38,115 +38,45 @@ steps:
       - package-it
       - integration-ess
     steps:
-      - label: "Default"
-        key: stateful-ubuntu-default-sudo
+      - label: "x86:Group: {{matrix}}"
```
Comment (rowlandgeoff) [inline .buildkite/bk.integration.pipeline.yml:41]: nit: more accurate to call this `x64`? Or `x86-x64`?

**Comment by rowlandgeoff** [comment]
> Run the new pipeline on push to main.

Do you want to enable with this PR or a follow-up?

File: .buildkite/bk.integration.pipeline.yml (hunk 1 above)
Comment (rowlandgeoff) [reply .buildkite/bk.integration.pipeline.yml:41]: Let's go with x86_64, since I think we use that term more consistently in other places

**Review by rowlandgeoff — APPROVED** [review]

**Review by leehinman — APPROVED** [review]
LGTM

**Pull request merged**